### PR TITLE
[bug-fix] Enable configurability for SCIM custom user schema URI

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose

$subject

### Related Issues

- https://github.com/wso2/product-is/issues/19432

### Related PRs

- Reverted PR: https://github.com/wso2/identity-apps/pull/5847

### Checklist

- [ ] e2e cypress tests locally verified. (for internal contributers)
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: apps/console/java/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2 [hunk 1]
```diff
@@ -1852,6 +1852,9 @@
         {% if console.enable_identity_claims is defined %}
         "enableIdentityClaims": {{ console.enable_identity_claims }},
         {% endif %}
+        {% if scim2.custom_user_schema_uri is defined %}
```
Comment (pavinduLakshan) [inline apps/console/java/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2:1855]: Why are we adding this to console deployment config even though it's not being used in the console app?

File: features/admin.users.v1/constants/user-management-constants.ts [hunk 2]
```diff
@@ -93,10 +93,6 @@ export class UserManagementConstants {
     public static readonly SCIM2_USER_SCHEMA: string = "urn:ietf:params:scim:schemas:core:2.0:User";
     public static readonly BULK_REQUEST_SCHEMA: string = "urn:ietf:params:scim:api:messages:2.0:BulkRequest";
 
-    // Schema related constants.
-    public static readonly ENTERPRISESCHEMA: string = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User";
```
Comment (pavinduLakshan) [inline features/admin.users.v1/constants/user-management-constants.ts:97]: Hope this wouldn't affect the saas deployment :)

**Review by pavinduLakshan — COMMENTED** [review]
Let's add changeset and fix lint issues

**Comment by JayaShakthi97** [comment]
@pavinduLakshan, will let you know once PR is ready to review. Marking as drafted.

File: apps/console/java/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2 (hunk 1 above)
Comment (JayaShakthi97) [reply apps/console/java/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2:1855]: Usage: https://github.com/wso2/identity-apps/pull/7213/files#diff-d7687dc53dbafb6c7c2e39bb220798613f42f1e6c7540f52db7d76e63a1f9c30R361

File: features/admin.users.v1/constants/user-management-constants.ts (hunk 2 above)
Comment (JayaShakthi97) [reply features/admin.users.v1/constants/user-management-constants.ts:97]: Verified

**Comment by JayaShakthi97** [comment]
Closing as per the comment: https://github.com/wso2/product-is/issues/19432#issuecomment-2550441153

**Pull request closed without merging**